libnavigation-core/src/main/java/com/mapbox/navigation/core/RoutesSetCallback.kt

```diff
@@ -0,0 +1,107 @@
+package com.mapbox.navigation.core
+
+import com.mapbox.navigation.base.route.NavigationRoute
+
+/**
+ * Interface definition for a callback that gets notified whenever routes
+ * passed to [MapboxNavigation.setNavigationRoutes] are set or produce an error and ignored.
+ */
+interface RoutesSetCallback {
+
+    /**
+     * Invoked whenever the routes passed to [MapboxNavigation.setNavigationRoutes]
+     * are successfully set.
+     *
+     * @param result [RoutesSetCallbackSuccess] object that describes the set routes
+     */
+    fun onRoutesSetResult(result: RoutesSetCallbackSuccess)
+
+    /**
+     * Invoked whenever the routes passed to [MapboxNavigation.setNavigationRoutes]
+     * produce an error and are ignored.
+     *
+     * @param result [RoutesSetCallbackError] object that describes the failure.
+     */
+    fun onRoutesSetError(result: RoutesSetCallbackError)
+}
```

Comment:
How about this instead:
```kotlin
fun interface RoutesSetCallback {
    fun onRoutesSet(result: Expected<RoutesSetError, RoutesSetSuccess>)
}
```

Much more concise and easier to consume from Kotlin (and Java as well).

Reply:
@LukasPaczos I think we don't expose `Expected`, do we?

Reply:
We do, it's a viable public API. A lot of Common SDK, Maps SDK, as well as our Nav UI components rely on it, less so Nav Core components but that's not a rule (we probably just had less opportunity to return result-based callbacks in the past).